usl_models/usl_models/flood_ml/batch_predictor.py

```diff
@@ -0,0 +1,191 @@
+            study_area=study_area_id,
+            city_cat_config=parsed_config_id,
+            batch_size=self.batch_size,
+            storage_client=self.client,
+        )
+        chunk_ids = []
+        for results in self.model.batch_predict_n(dataset, n=rainfall_duration):
+            for result in results:
+                chunk_id = result["chunk_id"]
+                if isinstance(chunk_id, tf.Tensor):
+                    chunk_id = chunk_id.numpy().decode("utf-8")
+
+                chunk_ids.append(chunk_id)
+                self.save_prediction_npy(
+                    study_area_id=study_area_id,
+                    config_id=config_id,
+                    chunk_id=chunk_id,
+                    prediction=result["prediction"].numpy(),
+                )
+
+        chunk_metadata = self.get_chunk_metadata(
+            study_area_id=study_area_id,
+            config_id=config_id,
+            chunk_ids=chunk_ids,
+        )
+
+        self.set_model_run_prediction_metadata(
+            study_area_id=study_area_id,
+            scenario_id=scenario_id,
+            chunk_metadata=chunk_metadata,
+        )
+
+        self.bundle_predictions_to_jsonl(
+            study_area_id=study_area_id,
+            config_id=config_id,
+            chunk_ids=chunk_ids,
+            scenario_id=scenario_id,
+        )
+
+    def set_model_run_metadata(self, scenario_ids: list[str]):
+        """Sets model run metadata in firestore."""
+        model_ref = self.db.collection("models").document(self.model_id)
+        run_ref = model_ref.collection("runs").document(self.run_id)
+        run_ref.set({"scenario_ids": scenario_ids})
+
+    def set_model_run_prediction_metadata(
+        self,
+        study_area_id: str,
+        scenario_id: str,
+        chunk_metadata: list[dict],
+    ):
+        """Sets the model run prediction metadata in firestore."""
+        model_ref = self.db.collection("models").document(self.model_id)
+        run_ref = model_ref.collection("runs").document(self.run_id)
+        prediction_id = f"Prediction-{study_area_id}-{scenario_id}"
+        prediction_ref = run_ref.collection("predictions").document(prediction_id)
+        prediction_ref.set(
+            {
+                "study_area_id": study_area_id,
+                "scenario_configuration_id": scenario_id,
+                "chunks": chunk_metadata,
+            }
+        )
+
+    def get_chunk_metadata(
+        self, study_area_id: str, config_id: str, chunk_ids: list[str]
+    ) -> list[dict]:
+        """Returns a list of chunk metadata."""
+        chunk_metadata = []
+        for chunk_id in chunk_ids:
+            path = self.get_prediction_npy_path(
+                study_area_id=study_area_id, config_id=config_id, chunk_id=chunk_id
+            )
+            chunk_metadata.append(
+                {
+                    "id": chunk_id,
+                    "path": f"gs://{self.output_bucket}/{path}",
```

Comment:
let's store the path to the JSONL file we generate which includes these predictions
those .npy files I think we should delete, so they won't be good to link to

Reply:
I've added a path at the model run level.

Reply:
I think it actually belongs here, because I don't think having one jsonl file for all predictions is an inherent part of the approach (I think we'd actually prefer to generate more smaller files so the export logic can start sooner.)

Reply:
While we are still using a single .jsonl file, do we want each chunk to point to the bundled file?

Reply:
I think so, because in the future we may (and hopefully will) have multiple jsonl files
I wouldn't want the database schema to overly reflect the current implementation